src/common/substrait/src/df_logical.rs

```diff
@@ -412,6 +476,24 @@ impl DFLogicalSubstraitConvertor {
     }
 }
 
+fn same_schema_without_metadata(lhs: &ArrowSchemaRef, rhs: &ArrowSchemaRef) -> bool {
+    lhs.fields.len() == rhs.fields.len()
+        && lhs.fields.iter().zip(rhs.fields.iter()).all(|(x, y)| {
+            x.name == y.name && x.data_type == y.data_type && x.is_nullable == y.is_nullable
+        })
+}
+
+fn try_convert_df_schema(df_schema: &DFSchemaRef) -> Result<Schema, Error> {
+    #[allow(clippy::needless_borrow)]
+    let arrow_schema: ArrowSchema = (&**df_schema).into();
+
+    let schema: Schema = arrow_schema
+        .try_into()
+        .map_err(BoxedError::new)
+        .context(InternalSnafu)?;
+    Ok(schema)
+}
```

Comment:
I prefer to implement this as `impl TryFrom<DfSchema> for Schema` like
```rust
impl TryFrom<Arc<ArrowSchema>> for Schema {
    type Error = Error;
```
which is more generic.

And by the way, can `df_schema.as_ref()` replace that `(&**df_schema)`? It's a trait method from `Arc` https://doc.rust-lang.org/std/sync/struct.Arc.html#impl-AsRef%3CT%3E-for-Arc%3CT%3E